### PR TITLE
Add time event constant

### LINKED ISSUE #1600: Provide constant that provides the maximum number of parallel time events of a state machine
In application scenarios that require implementations of timer services with static allocation of buffers this value will be relevant to set the correct dimensions. 

Currently the C++ code generator defines a constant for the overall number of time events used in a state chart. But as time events are often exclusive to each other this value is safe but may be too large.

### PR DESCRIPTION
**Pull request by BeckmaR**
Resolves #1600 

* Adds calculation for max number of parallel time events to sexec.StatechartExtensions
* Adds test for algorithm, including a new test statechart and an ExtensionsTestModels class
* Added as a `#define` to C generator and a static constant to C++

### CODE REVIEW

File: test-plugins/org.yakindu.sct.model.sexec.test/src/org/yakindu/sct/model/sexec/transformation/test/StatechartExtensionsTest.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import com.google.inject.Inject;
+
+import static org.junit.Assert.*;
+
+@RunWith(XtextRunner.class)
+@InjectWith(TestModelInjectorProvider.class)
+public class StatechartExtensionsTest {
+	@Inject protected StatechartExtensions statechartExtensions;
+	
+	@Test
```
Comment (terfloth) [inline test-plugins/org.yakindu.sct.model.sexec.test/src/org/yakindu/sct/model/sexec/transformation/test/StatechartExtensionsTest.java]: Add a test case for a stjatemachine without time events.

File: plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/StatemachineHeader.xtend [hunk 2]
```diff
@@ -212,6 +215,11 @@ class StatemachineHeader implements IContentTemplate {
 			/*! Define dimension of the state configuration vector for history states. */
 		#define «type.toUpperCase»_MAX_HISTORY_STATES «historyVector.size»«ENDIF»
 		
+		«IF timed»
+		/*! Define maximum number of time events that can be active at once */
+		#define «type.toUpperCase»_MAX_ACTIVE_TIME_EVENTS «(it.sourceElement as Statechart).maxNumberOfParallelTimeEvents»
```
Comment (terfloth) [inline plugins/org.yakindu.sct.generator.c/src/org/yakindu/sct/generator/c/StatemachineHeader.xtend]: - use naming extension as in c++ generator for calculating the constant name
- apply a consistent naming: 'ACTIVE' here vs 'parallel' everywhere else

File: test-plugins/org.yakindu.sct.model.sexec.test/src/org/yakindu/sct/model/sexec/transformation/test/StatechartExtensionsTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.yakindu.sct.model.sexec.transformation.test;
```
Comment (terfloth) [inline test-plugins/org.yakindu.sct.model.sexec.test/src/org/yakindu/sct/model/sexec/transformation/test/StatechartExtensionsTest.java]: Add copyright header to text class.

File: test-plugins/org.yakindu.sct.test.models/src/org/yakindu/sct/test/models/ExtensionTestModels.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.yakindu.sct.test.models;
```
Comment (terfloth) [inline test-plugins/org.yakindu.sct.test.models/src/org/yakindu/sct/test/models/ExtensionTestModels.java]: Add copyright header.

**Comment by BeckmaR** [comment]
All done.

**Pull request merged**